components/movementsensor/gpsrtkserial/gpsrtkserial.go

```diff
@@ -201,19 +201,16 @@ func newRTKSerial(
 	}
 
 	g.InputProtocol = serialStr
-	nmeaConf := &gpsnmea.Config{
-		ConnectionType: serialStr,
-	}
 
-	// Init NMEAMovementSensor
-	nmeaConf.SerialConfig = &gpsnmea.SerialConfig{
+	serialConfig := &gpsnmea.SerialConfig{
 		SerialPath:     newConf.SerialPath,
 		SerialBaudRate: newConf.SerialBaudRate,
 	}
-	g.nmeamovementsensor, err = gpsnmea.NewSerialGPSNMEA(ctx, conf.ResourceName(), nmeaConf, logger)
+	dev, err := gpsnmea.NewSerialDataReader(serialConfig, logger)
```

Comment:
This function should be moved from the gpsnmea namespace to rtkutils. In my mind, we should have no dependencies on gpsnmea at all, because that's the implementation of a different component and rtkutils is where the shared code goes.

Same feedback re: `NewI2cDataReader`